fsspec/implementations/local.py

```diff
@@ -232,10 +248,20 @@ def chmod(self, path, mode):
         return os.chmod(path, mode)
 
 
-def make_path_posix(path, sep=os.sep):
+def make_path_posix(path, sep=os.sep, remove_trailing_slash=False):
     """Make path generic"""
     if isinstance(path, (list, set, tuple)):
-        return type(path)(make_path_posix(p) for p in path)
+        return type(path)(
+            make_path_posix(p, remove_trailing_slash=remove_trailing_slash)
+            for p in path
+        )
+    path = _make_path_posix(path, sep)
+    if os.name == "nt" and len(path) == 3 and path[1] == ":":
+        return path  # windows root
+    return path.rstrip("/") if remove_trailing_slash else path
+
+
+def _make_path_posix(path, sep=os.sep):
```

Comment:
Why split up the function? The windows-root check is a rare one and should ideally be checked after more common things. This would be a second case within "# windows full path"?

Reply:
My reasoning was that the original `make_path_posix` function has several return statements. *Wrapping* the original function enables detection of special edge cases but also adds *support* for stripping the trailing suffix where appropriate.